cli/src/pcluster/commands.py

```diff
@@ -445,3 +446,99 @@ def update(args):
     except KeyboardInterrupt:
         LOGGER.info("\nExiting...")
         sys.exit(0)
+
+
+def delete_image(args):  # noqa: C901
+    """Delete image described by image_name."""
+    LOGGER.info("Deleting: %s", args.image_name)
+    LOGGER.debug("CLI args: %s", str(args))
+    try:
+        # delete image raises an exception if stack does not exist
+        result = PclusterApi().delete_image(image_name=args.image_name, region=utils.get_region(), force=args.force)
+        if isinstance(result, ImageBuilderInfo):
+            print(f"Image deletion started correctly. {result}")
+        else:
+            utils.error(f"Image deletion failed. {result.message}")
+
+        if result.stack_exist:
+            sys.stdout.write("\rImageBuilderStackStatus: %s" % result.stack_status)
+            sys.stdout.flush()
+            LOGGER.debug("ImageBuilderStackStatus: %s", result.stack_status)
+
+            if not args.nowait:
+                while result.stack_status == "DELETE_IN_PROGRESS":
```

Comment:
Try to look at utils.verify_stack_status I think you can reuse it or do something similar to simplify this funciton and remove the `# noqa: C901`

Reply:
Then I'm not sure we want to introduce this while/loop and nowait feature for image commands. I kept them for pcluster commands because they were already there and integ tests framework was relying on them but I think we can avoid this loop for these new commands.  
I think we'll remove the loop from the pcluster commands too.

Reply:
temporarily remove --no-wait